src/tools/rbd_wnbd/rbd_wnbd.cc

```diff
@@ -888,6 +940,9 @@ class RBDService : public ServiceBase {
         }
       }
 
+      std::thread adapter_monitor_thread(&monitor_wnbd_adapter, this);
+      adapter_monitor_thread.detach();
```

Comment:
> Here's the WMI cleanup procedure: https://learn.microsoft.com/en-us/windows/win32/wmisdk/cleaning-up-and-shutting-down-a-wmi-application. It doesn't really say what would happen if we skip closing WMI connections or calling CoUninitialize (which I'll have to add btw), I'm not sure if this can cause any leak on the wmi provider side, if it adds up to the wmi subscription quota or if it eventually gets cleaned up.

The reason I asked is that `WmiSubscription` and `WmiConnection` destructors would most likely never run since the "adapter monitor" thread is detached.  If this can cause a system-wide leak, this is obviously a problem.

> If `stop_hook` hangs or fails, Windows reports that it was unable to stop the service and it will continue to appear as running. Also, the service won't be restarted as part of the same process, it always uses a separate process.

Good -- at least there is no potential to get several of these "adapter monitor" threads ;)

Reply:
> The reason I asked is that WmiSubscription and WmiConnection destructors would most likely never run since the "adapter monitor" thread is detached. If this can cause a system-wide leak, this is obviously a problem.

Right. Well, just to be safe, I could avoid detaching the thread and actually wait for it before exiting.